### ISSUE TITLE
Trying to get md5 of local zip

### ISSUE THREAD

**Issue by jmdcal** [saltstack/salt#25983]
I am trying to unzip a local file
```saltstack
unzip_{{target.name}}:
    archive.extracted:
      - name:  "/projects/{{target.name}}/webapps/ROOT/"
      - source: "/home/jenkins/build/blah.zip
      - archive_format: zip
```
The thing is trying to get a hash:
```python-traceback
[DEBUG   ] Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/states/file.py", line 1558, in managed
    follow_symlinks)
  File "/usr/lib/python2.7/site-packages/salt/modules/file.py", line 4108, in manage_file
    dl_sum = get_hash(sfn, source_sum['hash_type'])
TypeError: string indices must be integers, not str

[ERROR   ] Unable to manage file: string indices must be integers, not str
[INFO    ] Completed state [/var/cache/salt/minion/files/dev/blah01_webapps_ROOT_.zip] at time 16:13:24.375870
```

Looks related to this code
https://github.com/saltstack/salt/blob/0216277dbb409b984e763265001a070f09a0ce58/salt/modules/file.py#L3794
@terminalmage

possible patch, works for me
```
            # If the downloaded file came from a non salt server or localsource                                                                                                               
            # verify that it matches the intended sum value                                                                                                                                   
            if _urlparse(source).scheme not in ( 'salt',''):
```

**Comment by benhosmer** [comment]
`salt-call state.apply test --log-level=debug`

The state file that triggers this:
```
geoserver:                                                                         
  file.managed:                                                                    
    - name: /var/lib/tomcat/webapps/geoserver.war                                  
    - source: /tmp/geoserver/geoserver.war
```

```
[DEBUG   ] Updating roots fileserver cache
[DEBUG   ] Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/salt/states/file.py", line 1545, in managed
    follow_symlinks)
  File "/usr/lib/python2.6/site-packages/salt/modules/file.py", line 3556, in manage_file
    dl_sum = get_hash(sfn, source_sum['hash_type'])
TypeError: string indices must be integers, not str
```

```
           Salt: 2015.5.3
         Python: 2.6.6 (r266:84292, Jan 22 2014, 09:42:36)
         Jinja2: unknown
       M2Crypto: 0.20.2
 msgpack-python: 0.4.6
   msgpack-pure: Not Installed
       pycrypto: 2.0.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.3.1
           RAET: Not Installed
            ZMQ: 3.2.5
           Mako: Not Installed
        Tornado: Not Installed
```

**Comment by cachedout** [comment]
Since we've merged that patch in #25984, I'll go ahead and close this. Please let me know if that's not correct. Thanks!

**Comment by edwardsgs** [comment]
I see this is closed, but I checked out this post while debugging an issue. The archive.extracted code in the original post has an unbalanced double quote on the "source" line. I wonder if that was part of the problem.

unzip_{{target.name}}:
    archive.extracted:
      - name:  "/projects/{{target.name}}/webapps/ROOT/"
      - source: "/home/jenkins/build/blah.zip
      - archive_format: zip
